Added sc-pipelines support

marcingrzejszczak opened the pull request.

marcingrzejszczak committed:
Added maven wrapper

marcingrzejszczak committed:
First contract test

marcingrzejszczak committed:
Backward compatibility of HAL is working - still in offline mode - need to change it to remote

marcingrzejszczak committed:
Added e2e and smoke

marcingrzejszczak committed:
Added distr manag

marcingrzejszczak committed:
Added missing version

marcingrzejszczak committed:
Migrated to main_module

marcingrzejszczak committed:
Changed service to postgresql

marcingrzejszczak committed:
Fixed wrong db type

marcingrzejszczak committed:
Fixed wrong service

marcingrzejszczak committed:
Fixed wrong service

marcingrzejszczak committed:
Tying to fix grant

marcingrzejszczak committed:
Tying to fix grant

marcingrzejszczak committed:
Added setup for smoke & e2e

marcingrzejszczak committed:
Fixed tests

marcingrzejszczak committed:
Fixed tests

marcingrzejszczak committed:
Removed spring runner

marcingrzejszczak committed:
Fixed tests:

marcingrzejszczak committed:
Fixed

marcingrzejszczak committed:
Turned on spring profile by default

marcingrzejszczak committed:
Fixed

pilloPl commented:
Thanks Marcin! Will take a look in the nearest future

michal-michaluk commented:
thx, cool, 3 points from me:

1. tests you ware forced to ignore are my fault, they are already fixed in another PR #11.
2. why you dropped jsonb type in postgres, any issue with it is supported since [postgresql 9.4 or even 9.3](https://www.postgresql.org/docs/9.4/static/datatype-json.html)
3. [travis build is currently failing](https://travis-ci.org/ddd-by-examples/factory/builds/387829371) at assembly plugin execution (empty assembly), shouldn't that assembly be defined in factory/app-monolith instead of root module factory/, app is packed there in uber jar?

marcingrzejszczak commented:
Ad. 2

michal-michaluk commented:
@marcingrzejszczak thx for info.
I will look on travis build, after addition of mvnw he picked some additional steps.
Could we have another liquidbase parameter value [activated via context = ENV or JVM property](http://www.liquibase.org/documentation/changelog_parameters.html) for Cloud Foundry's postrgesql?

marcingrzejszczak commented:
Sure

marcingrzejszczak committed:
Limited number of connections

marcingrzejszczak committed:
Conflicts

marcingrzejszczak commented:
I fixed the conflicts and added limiting of connections for cloud (for free plan of postrgresql I ran out of connections). Also I still have to ignore one test cause it's not passing for me

marcingrzejszczak committed:
Putting assembly descriptor to stub jar for tests to pass

pilloPl commented on adapter-commons/pom.xml:
```diff
@@ -41,9 +41,28 @@
     <properties>
-        <java.version>1.8</java.version>
+        <!-- [PIPELINE] -->
+        <distribution.management.release.id>artifactory-local</distribution.management.release.id>
```
localhost:8081? should be depend on hardcoded URL?

marcingrzejszczak replied:
You can override it and it happens via the pipeline

pilloPl commented on adapter-commons/src/main/resources/schema/commons.xml:
```diff
@@ -5,7 +5,7 @@
-    <changeSet author="jakubpilimon" id="2.postgres.json" dbms="postgresql">
-        <sql>CREATE CAST (character varying AS jsonb) WITH INOUT AS ASSIGNMENT</sql>
-    </changeSet>
+    <!--<changeSet author="jakubpilimon" id="2.postgres.json" dbms="postgresql">-->
```
commenting changes that might have already been run somewhere will cause problems :) I know you had problems with jsonb but you can at least add ignore on error for this changelog

marcingrzejszczak replied:
Feel free to fix it

pilloPl commented on app-monolith/src/main/resources/application.properties:
```diff
@@ -1,6 +1,7 @@
 spring.main.banner-mode=off
-spring.jpa.database=default
+spring.jpa.database=postgresql
```
postgres for default profile? What about running locally?

pilloPl commented on product-management-adapters/src/test/groovy/io/dddbyexamples/factory/product/management/ProductDescriptionPersistenceSpec.groovy:
```diff
@@ -1,11 +1,15 @@
 
 import static java.util.Collections.singletonList
 
+// TODO: Unignore
```
PR that fixes this test is already merged.

marcingrzejszczak replied:
It doesn't work on my machine

pilloPl commented on shortages-prediction-adapters/src/test/groovy/e2e/E2eSpec.groovy:
```diff
@@ -0,0 +1,41 @@
+
+	def 'should work'() {
+		expect:
+			Awaitility.await().atMost(this.timeout, TimeUnit.SECONDS).untilAsserted({
```
Why you use awaitility when in Spock you have PollingConditions?

marcingrzejszczak replied:
Because I copied it from a working JUnit test. Feel free to change it to whatever you want.

pilloPl commented on shortages-prediction-adapters/src/test/groovy/io/dddbyexamples/factory/shortages/prediction/monitoring/persistence/Config.groovy:
```diff
@@ -0,0 +1,45 @@
+			@Override
+			Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+				if (bean instanceof ShortagesDao) {
+					return Mockito.mock(ShortagesDao)
```
hmm, why?

marcingrzejszczak replied:
Because we want to mock out the access to the real database - https://github.com/spring-projects/spring-boot/issues/13298

marcingrzejszczak committed:
Fixing messed up rebase

marcingrzejszczak committed:
Came back to default for spring.jpa

marcingrzejszczak committed:
Added Gradle

marcingrzejszczak committed:
Removed awaitlity

marcingrzejszczak committed:
Moving to Finchley RC2 and 2.0.0.RC2 for sc-contract

marcingrzejszczak committed:
Revert "Moving to Finchley RC2 and 2.0.0.RC2 for sc-contract"

This reverts commit 6152fa2db7cb608858811cf66a132c01d3dc15f4.

marcingrzejszczak committed:
Trying to make this work on concourse

marcingrzejszczak committed:
Trying to make this work on concourse

marcingrzejszczak committed:
Moved the allow snapshot section out

marcingrzejszczak committed:
Moved the allow snapshot section out
